adapters/repos/db/vector/hnsw/compress.go

```diff
@@ -50,9 +50,15 @@ func (h *hnsw) Compress(cfg ent.PQConfig) error {
 
 	dims := int(h.dims)
 
-	// segments == 0 (default value) means use as many segments as dimensions
 	if cfg.Segments <= 0 {
-		cfg.Segments = dims
+		for i := 6; i > 0; i-- {
+			if dims%i == 0 {
+				cfg.Segments = dims / i
+				h.pqConfig.Segments = cfg.Segments
```

Comment:
are we sure this change is persisted and valid after db restart?

Reply:
Should be safe like [this](https://github.com/weaviate/weaviate/blob/c6f516b198809d0ea88713d9a77da1f0760061b4/adapters/repos/db/vector/ssdhelpers/product_quantization.go#L221)